packages/common/src/ee/AiAgent/schemas/parser.ts

```diff
@@ -0,0 +1,60 @@
+import assertUnreachable from '../../../utils/assertUnreachable';
+import {
+    toolDashboardArgsSchema,
+    toolFindChartsArgsSchema,
+    toolFindDashboardsArgsSchema,
+    toolFindExploresArgsSchemaV1,
+    toolFindExploresArgsSchemaV2,
+    toolFindFieldsArgsSchema,
+    toolImproveContextArgsSchema,
+    toolProposeChangeArgsSchema,
+    toolRunQueryArgsSchema,
+    toolSearchFieldValuesArgsSchema,
+    toolTableVizArgsSchema,
+    toolTimeSeriesArgsSchema,
+    toolVerticalBarArgsSchema,
+} from './tools';
+import { type ToolName } from './visualizations';
+
+/**
+ * Parse the tool args using the specific schema for each tool name
+ * @param toolName - The name of the tool to parse
+ * @param toolArgs - The tool args to parse
+ * @returns The parsed tool args
+ */
+export const parseToolArgs = (toolName: ToolName, toolArgs: unknown) => {
+    switch (toolName) {
+        case 'findExplores':
+            // Try v2 first, fallback to v1
+            const v2Result = toolFindExploresArgsSchemaV2.safeParse(toolArgs);
+            if (v2Result.success) return v2Result;
+            return toolFindExploresArgsSchemaV1.safeParse(toolArgs);
+        case 'findFields':
+            return toolFindFieldsArgsSchema.safeParse(toolArgs);
+        case 'searchFieldValues':
+            return toolSearchFieldValuesArgsSchema.safeParse(toolArgs);
+        case 'generateBarVizConfig':
+            return toolVerticalBarArgsSchema.safeParse(toolArgs);
+        case 'generateTableVizConfig':
+            return toolTableVizArgsSchema.safeParse(toolArgs);
+        case 'generateTimeSeriesVizConfig':
+            return toolTimeSeriesArgsSchema.safeParse(toolArgs);
+        case 'generateDashboard':
+            return toolDashboardArgsSchema.safeParse(toolArgs);
+        case 'findDashboards':
+            return toolFindDashboardsArgsSchema.safeParse(toolArgs);
+        case 'findCharts':
+            return toolFindChartsArgsSchema.safeParse(toolArgs);
+        case 'improveContext':
+            return toolImproveContextArgsSchema.safeParse(toolArgs);
+        case 'proposeChange':
+            return toolProposeChangeArgsSchema.safeParse(toolArgs);
+        case 'runQuery':
+            return toolRunQueryArgsSchema.safeParse(toolArgs);
+        default:
+            return assertUnreachable(
+                toolName,
+                `Unknown tool name: ${toolName}`,
+            );
+    }
+};
```

Comment:
IIRC we were using base schemas because of the zod limitation: transformed schemas cannot be in a discriminated union.

In this case, we can start using transformed schemas because they map to our domain types

Reply:
I see! 
Now transformed schemas are used on this parser util.

Note: `toolImproveContextArgsSchema` and `toolProposeChangeArgsSchema` don't have transformed versions (they don't need transformations), so they remain as is.